python-checks/src/main/java/org/sonar/python/checks/cdk/IamPolicyPublicAccessCheck.java

```diff
@@ -0,0 +1,161 @@
+import static org.sonar.python.checks.cdk.CdkUtils.getArgument;
+import static org.sonar.python.checks.cdk.CdkUtils.getCall;
+import static org.sonar.python.checks.cdk.CdkUtils.getDictionary;
+import static org.sonar.python.checks.cdk.CdkUtils.getDictionaryPair;
+import static org.sonar.python.checks.cdk.CdkUtils.getListExpression;
+
+@Rule(key = "S6270")
+public class IamPolicyPublicAccessCheck extends AbstractCdkResourceCheck {
+
+  private static final String ISSUE_MESSAGE = "Make sure granting public access is safe here.";
+  private static final String SECONDARY_MESSAGE = "Access is set to \"ALLOW\" here.";
+
+  private static final String IAM_EFFECT_ALLOW = "aws_cdk.aws_iam.Effect.ALLOW";
+
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqn("aws_cdk.aws_iam.PolicyStatement", IamPolicyPublicAccessCheck::checkPolicyStatementConstructor);
+    checkFqn("aws_cdk.aws_iam.PolicyStatement.from_json", IamPolicyPublicAccessCheck::checkPolicyStatementJson);
+    checkFqn("aws_cdk.aws_iam.PolicyDocument.from_json", IamPolicyPublicAccessCheck::checkPolicyDocumentJson);
+  }
+
+  private static void checkPolicyStatementConstructor(SubscriptionContext subscriptionContext, CallExpression callExpression) {
+    // The rule should only trigger if "effect" is set to "ALLOW"
+    Optional<Expression> sensitiveEffect = getSensitiveEffect(subscriptionContext, callExpression);
+    if (sensitiveEffect.isEmpty()) {
+      return;
+    }
+
+    getArgument(subscriptionContext, callExpression, "principals")
+      .flatMap(CdkUtils::getListExpression)
+      .ifPresent(listLiteral -> listLiteral.elements().expressions().stream()
+        .filter(isSensitivePrincipal())
+        .forEach(sensitivePrincipal -> raiseIssue(subscriptionContext, sensitivePrincipal, sensitiveEffect.get())));
+  }
+
+  private static void checkPolicyStatementJson(SubscriptionContext subscriptionContext, CallExpression callExpression) {
+    if (callExpression.arguments().isEmpty()) {
+      return;
+    }
+
+    getArgument(subscriptionContext, callExpression, "obj", 0)
+      .flatMap(e -> e.getExpression(e2 -> e2.is(Tree.Kind.DICTIONARY_LITERAL)))
+      .map(DictionaryLiteral.class::cast)
+      .ifPresent(dictionaryLiteral -> checkJsonEntry(subscriptionContext, dictionaryLiteral));
+  }
+
+  private static void checkPolicyDocumentJson(SubscriptionContext subscriptionContext, CallExpression callExpression) {
+    Optional<DictionaryLiteral> dictionaryLiteral = getArgument(subscriptionContext, callExpression, "obj", 0)
+      .flatMap(e -> e.getExpression(isDictionaryLiteral()))
+      .map(DictionaryLiteral.class::cast);
+
+    if (dictionaryLiteral.isEmpty()) {
+      return;
+    }
+
+    Optional<ListLiteral> statement = CdkUtils.getDictionaryPair(subscriptionContext, dictionaryLiteral.get(), "Statement")
+      .flatMap(resolvedKeyValuePair -> resolvedKeyValuePair.value.getExpression(isListLiteral()))
+      .map(ListLiteral.class::cast);
```

Comment:
```suggestion
      .flatMap(CdkUtils::getList)
```
Same for this helper